trollflow2/tests/test_trollflow2.py

```diff
@@ -2015,5 +2049,88 @@ def fake_persist(*args):
     assert not sc_3a_3b["another"].attrs.get("persisted")
 
 
+yaml_test_s3_uploader_plain = """
+product_list:
+  output_dir: /tmp/
+  publish_topic: /topic
+  s3_config:
+    target: s3://bucket-name/
+  fname_pattern:
+    "{start_time:%Y%m%d_%H%M}_{platform_name}_{areaname}_{productname}.{format}"
+
+  areas:
+    euro4:
+      areaname: euro4
+      products:
+        airmass:
+          productname: airmass
+          formats:
+          - format: tif
+            writer: geotiff
+        natural_with_ir:
+          productname: natural_with_colorized_ir_clouds
+          formats:
+          - format: tif
+            writer: geotiff
+
+"""
+
+
+def test_s3_uploader_update_filenames():
+    """Ensure that filenames are updated when transfer is made to S3."""
+    from yaml import UnsafeLoader
+    from trollflow2.plugins import s3_uploader
+    from trollflow2.dict_tools import plist_iter
+
+    product_list = read_config(raw_string=yaml_test_s3_uploader_plain, Loader=UnsafeLoader)
+    job = {"product_list": product_list, "input_mda": input_mda.copy()}
+    _ = _create_filenames_and_topics(job)
+
+    with mock.patch('trollflow2.plugins.S3Mover'):
+        s3_uploader(job)
+    for fmt, _ in plist_iter(job['product_list']['product_list']):
+        assert fmt['filename'].startswith('s3://bucket-name/')
+
+
+def test_s3_uploader_copy():
+    """Test that S3 mover is only copying the file."""
+    from yaml import UnsafeLoader
+    product_list = read_config(raw_string=yaml_test_s3_uploader_plain, Loader=UnsafeLoader)
+    job = {"product_list": product_list, "input_mda": input_mda.copy()}
+    _ = _create_filenames_and_topics(job)
+
+    with mock.patch('trollflow2.plugins.S3Mover') as S3Mover:
+        from trollflow2.plugins import s3_uploader
+
+        s3_uploader(job)
+
+        assert mock.call(
+            "/tmp/20190217_0600_NOAA-15_euro4_airmass.tif",
```

Comment:
Should this use `tmp_path`?

Reply:
No, the user has defined `/tmp/` as the output directory. Nothing is written here, we just check that the mover is called with correct values.